hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraphBuilder.java

```diff
@@ -2,25 +2,23 @@
 
 import java.time.Duration;
 import java.time.Instant;
-import java.util.concurrent.TimeUnit;
 import org.hypertrace.core.datamodel.StructuredTrace;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ApiTraceGraphBuilder {
   private static final Logger LOG = LoggerFactory.getLogger(ApiTraceGraphBuilder.class);
 
-  private static ThreadLocal<ApiTraceGraph> cachedGraph = new ThreadLocal<>();
-  private static ThreadLocal<StructuredTrace> cachedTrace = new ThreadLocal<>();
+  private static final ThreadLocal<ApiTraceGraph> cachedGraph = new ThreadLocal<>();
+  private static final ThreadLocal<StructuredTrace> cachedTrace = new ThreadLocal<>();
 
   public static ApiTraceGraph buildGraph(StructuredTrace trace) {
-    if (!GraphBuilderUtil.isSameStructuredTrace(cachedTrace.get(), trace)) {
+    if (GraphBuilderUtil.isStructuredTraceChanged(cachedTrace.get(), trace)) {
```

Comment:
I thin, ApiTraceGraph is independent of entitygraph. So, do we have to built it again?

Reply:
Yeah, I observed that. But thought of taking up that refactor separately.